src/puppetlabs/jdbc_util/core.clj

```diff
@@ -165,3 +165,22 @@
       first
       :count
       pos?))
+
+(defn get-sequence-name
+  "Returns the name of the sequence associated with a column, or nil if there is
+  no sequence."
+  [db table column]
+  (-> (jdbc/query db ["SELECT pg_get_serial_sequence(?, ?)" table column])
+      first
+      :pg_get_serial_sequence))
+
+(defn reconcile-sequence!
+  "Finds the sequence associated with the given column and sets it to the
+  current maximum value in that column, so that the next value in the sequence
+  will be the current maximum + 1 (or 0, if the table is empty). If the column
+  has no associated sequence, throws an IllegalArgumentException."
+  [db table column]
+  (if-let [sequence-name (get-sequence-name db table column)]
+    (jdbc/query db [(format "SELECT setval('%s', (SELECT COALESCE(max(\"%s\"), 0) FROM \"%s\"))"
+                            sequence-name column table)])
+    (throw (IllegalArgumentException. (format "No sequence found for column %s on table %s." table column)))))
```

Comment:
Yeah, I wasn't sure what to use. I didn't want to forge a `PSQLException`, but I guess I could use `ExceptionInfo` or even just `Exception`?